python/setup.py

```diff
@@ -52,6 +52,13 @@ def get_cuda_version_from_header(cuda_include_dir):
 cuda_lib_dir = os.path.join(CUDA_HOME, "lib64")
 CUDA_VERSION = get_cuda_version_from_header(cuda_include_dir)
 
+INSTALL_PREFIX = os.environ.get("INSTALL_PREFIX", False)
+if not os.path.isdir(INSTALL_PREFIX):
+    raise OSError(
+        f"Invalid INSTALL_PREFIX: directory does not exist: {INSTALL_PREFIX}"
+    )
+rmm_include_dir = os.path.join(INSTALL_PREFIX, "include")
```

Comment:
Not everyone uses the `build.sh` so we can't rely on the `INSTALL_PREFIX` being set when someone runs `python setup.py`

Reply:
So the idea behind using the rmm "install" (even it may just be in a subdir of the build directory) is that the python modules are built against not only the rmm headers (which are the same as the ones in source dir, so this doesn't really make a difference), but also against other thirdparty dependencies that RMM pulled in via FetchContent, like spdlog or, eventually, Thrust.

This is, admittedly, never going to be perfect. In the case of pre-installed dependencies, there's no guarantee that the installed version of spdlog that cmake detects is going to be the same that ends up being used in the python build. Though it'll likely work in most standard situations, except for the case covered here, ie., cmake downloaded spdlog and the python build would have no idea about that without this PR.

I guess the best I can come up with to fix your concern is to handle the case of `INSTALL_PREFIX` not being set above and falling back to the previous hard-coded relative path. How's that?

Reply:
> This is, admittedly, never going to be perfect. In the case of pre-installed dependencies, there's no guarantee that the installed version of spdlog that cmake detects is going to be the same that ends up being used in the python build. Though it'll likely work in most standard situations, except for the case covered here, ie., cmake downloaded spdlog and the python build would have no idea about that without this PR.

It's on my TODO list to explore using [scikit-build](https://scikit-build.readthedocs.io/en/latest/) for handling building the Python extensions. This should allow us to reuse the existing CMake machinery we've already built and make the native code dependency handling much cleaner from the Python side.

> I guess the best I can come up with to fix your concern is to handle the case of INSTALL_PREFIX not being set above and falling back to the previous hard-coded relative path. How's that?

That makes sense to me!

Reply:
Okay, put the fallback back in.

I didn't really know about scikit-build, but at least from a quick look at their docs, it does seem appealing ;)